react-backend/client/src/components/visitors/qrcode.js

```diff
@@ -53,58 +63,72 @@ export class QRCode extends Component {
       username: '',
       qrcode: '',
       activity: 'not selected',
-      activities: []
-    }
+      activities: [],
+    };
 
     this.handleVideo = this.handleVideo.bind(this);
 
     this.changeActivity = this.changeActivity.bind(this);
   }
 
-  handleVideo = (e) => {
+  handleVideo = e => {
     let newState = {};
-    newState['login'] = this.state.login+1;
+    newState['login'] = this.state.login + 1;
     this.setState(newState);
   };
 
-  changeActivity = (newActivity) => {
+  changeActivity = newActivity => {
     this.setState({
-      activity: newActivity
+      activity: newActivity,
     });
 
     const visitInfo = {
       hash: this.state.hash,
-      activity: newActivity
-    }
+      activity: newActivity,
+    };
 
     return fetch('/postActivity', {
-      method: "POST",
-      body: JSON.stringify(visitInfo)
-    }).then(
-      this.props.history.push('/visitor/end')
-    );
+      method: 'POST',
+      body: JSON.stringify(visitInfo),
+    })
+      .then(this.props.history.push('/visitor/end'))
+      .catch(error => {
+        console.log('ERROR HAPPENING AT FETCH /postActivity', error);
+      });
   };
 
-  componentDidMount(){
-    if(this.state.login===1){
+  componentDidMount() {
+    if (this.state.login === 1) {
       instascan()
-      .then(getUserFromQRScan)
-      .then((user)=>{
-        this.setState({username:user.fullname.fullname, hash:user.fullname.hash});
-      })
+        .then(content => {
+          this.handleVideo();
+          return getUserFromQRScan(content);
+        })
+        .then(user => {
+          this.setState({
+            username: user.fullname.fullname,
+            hash: user.fullname.hash,
+          });
+        })
+        .catch(error => {
+          console.log('ERROR HAPPENING AT INSTASCAN', error);
+        });
     }
 
     fetch('/activities')
-    .then((res)=>res.json())
-    .then((res)=>res.activities)
-    .then((activities)=> {
-      this.setState({activities})
-    })
+      .then(res => res.json())
+      .then(res => res.activities)
+      .then(activities => {
+        this.setState({ activities });
+      })
+      .catch(error => {
+        console.log('ERROR HAPPENING AT FETCH', error);
```

Comment:
if there is a problem fetching activities, it may be worth sending the user back to the start of the login flow, or sending them to `/visitor/qrerror`

Reply:
@Zooey, thank you a lot! added on line 125